app/js/lib/foxbox.js

```diff
@@ -192,21 +197,44 @@ export default class Foxbox extends Service {
   /**
    * Start or stop polling.
    *
-   * @param {boolean} pollingEnabled
+   * @param {boolean} pollingEnabled Flag that indicates whether polling should
+   * be started or stopped.
    */
-  togglePolling(pollingEnabled = settings.pollingEnabled) {
-    if (pollingEnabled) {
-      this.pollingInterval = setInterval(this.refreshServicesByPolling.bind(this),
-        settings.pollingInterval);
+  togglePolling(pollingEnabled) {
+    this[priv.isPollingEnabled] = pollingEnabled;
 
-      // We immediately sync the data with the box.
-      this.refreshServicesByPolling();
+    if (pollingEnabled) {
+      this.schedulePoll();
     } else {
-      clearInterval(this.pollingInterval);
+      // Cancel next poll attempt if it has been scheduled.
+      clearTimeout(this[priv.nextPollTimeout]);
+      this[priv.nextPollTimeout] = null;
     }
   }
 
   /**
+   * Schedules an attempt to poll the server, does nothing if polling is not
+   * enabled or it has already been scheduled. New poll is scheduled only once
+   * previous one is completed or failed.
+   */
+  schedulePoll() {
+    // Return early if polling is not enabled or it has already been scheduled.
+    if (!this[priv.isPollingEnabled] || this[priv.nextPollTimeout]) {
+      return;
+    }
+
+    this[priv.nextPollTimeout] = setTimeout(() => {
+      this.refreshServicesByPolling().catch((e) => {
+        console.error('Polling has failed, scheduling one more attempt: ', e);
+      }).then(() => {
```

Comment:
Please add a line break before `.then(...`.
(See the rest of the code for coding styles)